analyses/tmb-compare-tcga/TMB_d3b_code/code/01_calculate_tmb_targetflexible_withbothexperstrt_and_cohort.py

```diff
@@ -0,0 +1,235 @@
+    for line in in_bed:
+        if line.split()[0] != "chrom":
+            try:
+                total_length += int(line.split()[2]) - int(line.split()[1])
+            except IndexError:
+                print("Check BED file formatting\n")
+    return total_length
+
+
+############### REading in input files #################################
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-i", "--maf", required=True, help="path to the MAF file")
+parser.add_argument(
+    "-m", "--metadatafile", required=True, help="path to the metadata/histology file"
+)
+parser.add_argument("-o", "--outfilename", required=True, help="Out file name")
+parser.add_argument(
+    "-c",
+    "--configfile",
+    required=True,
+    help="calculate_tmb.cfg.txt file with columns for disease, samplename, variant types etc.",
+)
+parser.add_argument(
+    "-w",
+    "--targetconfig",
+    required=True,
+    help="File with experimental strategy  and path to BED file",
+)
+args = parser.parse_args()
+
+
+################################################################
+# Defining MAF columns and metadata spcific fields from config #
+
+needed_cols = [
+    "Chromosome",
+    "Start_Position",
+    "End_Position",
+    "Variant_Classification",
+    "Tumor_Sample_Barcode",
+]
+
+with open(args.configfile) as configlines:
+    for line in configlines.readlines():
+        if line.startswith("Variants_to_use_for_TMB="):
+            var_class = line.split("=")[1].split(",")
+        if line.startswith("disease_column="):
+            disease_col = line.split("=")[1].strip("\n")
+        if line.startswith("samplename_column="):
+            samplename_col = line.split("=")[1].strip("\n")
+        if line.startswith("cohort_column="):
+            cohort_col = line.split("=")[1].strip("\n")
+        if line.startswith("typeoftargetcolumn="):
+            typeoftargetcol = line.split("=")[1].strip("\n")
+
+
+###########################################################
+
+
+####### Preparing MAF file ##################################
+# Loading MAF file
+maf_file = pd.read_table(args.maf, na_values=["."], comment="#", sep="\t")
+maf_file = maf_file.drop_duplicates(keep=False)
+allsamples = np.unique(maf_file["Tumor_Sample_Barcode"])
+
+# Filter  MAFbased on columns and also based on "variant_classification col"
+maf_file = maf_file[needed_cols]
+
+maf_file = maf_file.loc[
+    maf_file.apply(lambda x: x["Variant_Classification"] in var_class, axis=1)
+]
+filteredsamples = np.unique(maf_file["Tumor_Sample_Barcode"])
+###########################################################
+
+
+########### Preparing target files ########################
+target_dict = get_target_dict(args.targetconfig)
+############################################################
+
+
+############  Groupby and calculate TMB #####################
+outfile = open(args.outfilename, "w")
+outfile.write("Samplename\texperimental_strategy\tcohort\tdisease\tcount\tbedlength\tTMB\n")
```

Comment:
I may not be following this entirely, so corrrect me if I'm wrong, but if possible it would be nice to have the column names stay consistent with their origin files so it is easier to follow this back. So `Tumor_Sample_Barcode` again for example. (Sorry to keep harping on this same thing...)

Reply:
The initial idea was for this to be pretty generic. But since MAF file has this header as `Tumor_Sample_Barcode`, using the same in this output file is a good idea!